Check for existance of xterm & xclip commands.
Select first available utf-8 locale or fail if there isn't any.
Use colorit with its default configuration.

Index: dictd/dict_lookup
===================================================================
--- dictd.orig/dict_lookup	2011-02-02 20:17:55.000000000 +0100
+++ dictd/dict_lookup	2011-02-02 20:18:01.000000000 +0100
@@ -30,7 +30,7 @@
 D_AWK_PROG=${D_AWK_PROG-awk}
 D_XCLIP_PROG=${D_XCLIP_PROG-xclip}
 D_XTERM_PROG=${D_XTERM_PROG-xterm}
-D_UTF8_LOCALE=${D_UTF8_LOCALE-en_US.UTF-8}
+D_UTF8_LOCALE=${D_UTF8_LOCALE-`locale -a |fgrep .utf8 | head -n 1`}
 D_DICT_PROG=${D_DICT_PROG-dict}
 D_DICT_SERVER=${D_DICT_SERVER--h dict.org}
 D_DICT_PORT=${D_DICT_PORT--p 2628}
@@ -42,16 +42,16 @@
 D_TERM_X=${D_TERM_X-+200}
 D_TERM_Y=${D_TERM_Y-+200}
 D_TERM_H=${D_TERM_H-+40}
-D_PAGER=${D_PAGER-less}
+D_PAGER=${D_PAGER-less -R}
 
-if test -z "$COLORIZER"; then
-    colorit_config=$HOME/.colorit.d/dict
-    if test -f "$colorit_config"; then
-	COLORIZER="colorit -P - -c $colorit_config"
-    else
-	COLORIZER=cat
-    fi
-fi
+#if test -z "$COLORIZER"; then
+#    colorit_config=$HOME/.colorit.d/dict
+#    if test -f "$colorit_config"; then
+	COLORIZER="colorit" # -P - -c $colorit_config"
+#    else
+#	COLORIZER=cat
+#    fi
+#fi
 
 ############################################################
 # functions
@@ -81,6 +81,33 @@
     fi
 }
 
+check_system() {
+  err=0
+  for cmd in $D_DICT_CMD $D_XCLIP_PROG; do
+  	if ! which "$cmd" > /dev/null; then
+		missing="$missing $cmd"
+	fi
+  done
+
+  if [ -n "$missing" ] ; then
+	echo "Cannot find the following program(s) in the PATH: $missing"
+	echo "Please install the appropriate package(s)."
+	echo
+	err=1
+  fi
+
+
+  if [ -z "$D_UTF8_LOCALE" ] ; then
+	echo "Cannot find any UTF-8 locale installed on system."
+	echo "Please run \`dpkg-reconfigure locales' and choose some UTF-8 locale."
+	echo
+	err=1
+ fi
+
+ [ "$err" = 0 ] || exit $err;
+}
+
+
 ############################################################
 # main
 #
@@ -109,11 +136,13 @@
 done
 shift `expr $OPTIND - 1`
 
+check_system
+
 set_query
 
-tmp_dir="/tmp/`basename $0`.$$"
+tmp_dir=`mktemp -d`
+test -d "$tmp_dir" || exit 1;
 trap "rm -rf $tmp_dir" 0 1 2 3 15
-mkdir -m 700 "$tmp_dir"
 
 if test -z "$D_DICT_CMD"; then
     D_DICT_CMD="$D_DICT_PROG $D_DICT_SERVER $D_DICT_PORT $D_DICT_DB_OPT"
Index: dictd/dict_lookup.1.in
===================================================================
--- dictd.orig/dict_lookup.1.in	2011-02-02 20:17:55.000000000 +0100
+++ dictd/dict_lookup.1.in	2011-02-02 20:19:02.000000000 +0100
@@ -52,7 +52,7 @@
 .TP
 .B D_UTF8_LOCALE
 UTF-8 locale.
-.RI [ en_US.UTF-8 ]
+.RI [ "first available UTF-8 locale on system" ]
 .TP
 .B D_DICT_PROG
 Dictionary client program.
@@ -85,6 +85,7 @@
 .B D_TEXT_WIDTH
 Program calculating text width for given text.
 .RI [ cyrillic_text_width ]
+.br
 cyrillic_text_width doesn't work correctly for Japanese, Korean, Chinese
 and some other languages.
 .TP
@@ -102,7 +103,7 @@
 .TP
 .B D_PAGER
 Pager.
-.RI [ less ]
+.RI [ "less -R" ]
 .SH "SEE ALSO"
 .BR dict (1),
 .BR less (1),
